src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -644,6 +651,25 @@ private AnomalyDetectorSettings() {}
     // expensive maintenance (e.g., queue maintenance) with 1/10000 probability
     public static final int MAINTENANCE_FREQ_CONSTANT = 10000;
 
+    // ======================================
+    // Checkpoint setting
+    // ======================================
+    // we won't accept a checkpoint larger than 30MB. Or we risk OOM.
+    // For reference, in RCF 1.0, the checkpoint of a RCF with 50 trees, 10 dimensions,
+    // 256 samples is of 3.2MB.
+    // In compact rcf, the same RCF is of 163KB.
+    // Since we allow at most 5 features, and the default shingle size is 8 and default
+    // tree number size is 100, we can have at most 25.6 MB. It is possible that cx increases
+    // the max features or shingle size, but we don't want to risk OOM for the flexibility.
```

Comment:
Any experiments done on increasing these defaults with compact RCF, like higher default tree size or shingle size?

Reply:
There are some experiments here done by Sudipto/Josh: https://quip-amazon.com/A8hTAbgyHFtF/Random-Cut-Forest-serialization-benchmarks

Reply:
Ok, will take a look. As a reminder, should not have inaccessible links in public PRs 😃

Reply:
right, next time will send you offline.